src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -148,6 +149,9 @@ async Task EnsureLong()
                 }
                 catch (Exception e)
                 {
+                    if (e is OperationCanceledException)
```

Comment:
Note the special case for `dontBreakOnTimeouts` above - this behavior should be implemented for async as well.